src/unsafe_self_cell.rs

```diff
@@ -64,19 +64,29 @@ impl<ContainedIn, Owner, DependentStatic> UnsafeSelfCell<ContainedIn, Owner, Dep
         &(*joined_ptr.as_ptr()).dependent
     }
 
-    pub unsafe fn borrow_mut<'a, Dependent>(&'a mut self) -> &'a mut JoinedCell<Owner, Dependent> {
+    pub unsafe fn borrow_mut<'a, Dependent>(&'a mut self) -> (&'a Owner, &'a mut Dependent) {
```

Comment:
One could argue it's not part of the documented interface, so we are free to change it. Technically even #22 is a breaking change because one could have written code that observes the change (Hyrum's law).

Reply:
changes to `doc(hidden)` API is usually not considered a breaking change. This doesn’t require a major version bump.